dojo/importers/importer/importer.py

```diff
@@ -292,13 +295,48 @@ def import_scan(self, scan, scan_type, engagement, lead, environment, active, ve
         if cobaltio_config and cobaltio_config.product != engagement.product:
             raise ValidationError('"cobaltio_config" has to be from same product as "engagement"')
 
-        logger.debug('IMPORT_SCAN: Create Test')
-        test = self.create_test(scan_type, engagement, lead, environment, scan_date=scan_date, tags=tags,
-                            version=version, branch_tag=branch_tag, build_id=build_id, commit_hash=commit_hash, now=now,
-                            sonarqube_config=sonarqube_config, cobaltio_config=cobaltio_config)
-
-        logger.debug('IMPORT_SCAN: Parse findings')
-        parsed_findings = importer_utils.parse_findings(scan, test, active, verified, scan_type)
+        # check if the parser that handle the scan_type manage tests
+        # if yes, we parse the data first
+        # after that we customize the Test_Type to reflect the data
+        # This allow us to support some meta-formats like SARIF or the generic format
+        parser = get_parser(scan_type)
+        if hasattr(parser, 'get_tests'):
+            logger.debug('IMPORT_SCAN parser v2: Create Test and parse findings')
+            tests = parser.get_tests(scan_type, scan)
+            # for now we only consider the first test in the list and artificially aggregate all findings of all tests
+            # this is the same as the old behavior as current import/reimporter implementation doesn't handle the case
+            # when there is more than 1 test
+            #
+            # we also aggregate the label of the Test_type to show the user the original scan_type
+            # only if they are different. This is to support meta format like SARIF
+            # so a report that have the label 'CodeScanner' will be changed to 'SARIF > CodeScanner'
+            test_type_str = scan_type
```

Comment:
`test_type_name` would be more descriptive here as well.

Reply:
Changed.